python/cudnn/sdpa/fwd/api_dsl.py

```
@@ -983,8 +989,83 @@ def compile(self) -> None:
+        sched_policy = self._sched_policy()
+        try:
+            cfg, _ = make_cfg(self._template_params(sched_policy, False, 1))
+        except ValueError:
+            return 1
+        ctas_per_tile = int(getattr(cfg, "CTA_MMA", 1))
+        rows_per_tile = int(getattr(cfg, "TILES_Q", 1)) * int(cfg.TILE_M) * ctas_per_tile
+        dev = getattr(self.q_desc, "device", None)
+        ordinal = getattr(dev, "index", None)
+        try:
+            sm_count = device_info(torch.cuda.current_device() if ordinal is None else ordinal).sm_count
+        except Exception:  # no driver / odd device: never split rather than guess
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)